src/core/server/core_usage_data/core_usage_data_service.ts

```diff
@@ -256,6 +262,87 @@ export class CoreUsageDataService implements CoreService<CoreUsageDataSetup, Cor
     };
   }
 
+  private getMarkedAsSafe(
+    exposedConfigsToUsage: ExposedConfigsToUsage,
+    pluginId?: string,
+    usedPath?: string
+  ): { explicitlyMarked: boolean; isSafe: boolean } {
+    if (!pluginId || !usedPath) return { explicitlyMarked: false, isSafe: false };
+    const exposeDetails = exposedConfigsToUsage.get(pluginId);
+    if (!exposeDetails) {
+      return { explicitlyMarked: false, isSafe: false };
+    }
+
+    const exposeKeyDetails = Object.keys(exposeDetails).find((exposeKey) => {
+      const fullPath = `${pluginId}.${exposeKey}`;
+      const hasIntersection = hasConfigPathIntersection(usedPath, fullPath);
+
+      return hasIntersection;
+    });
+    if (!exposeKeyDetails) {
+      return { explicitlyMarked: false, isSafe: false };
+    }
+
+    const isSafe = exposeDetails[exposeKeyDetails];
+    if (typeof isSafe === 'boolean') {
+      return { explicitlyMarked: true, isSafe };
+    }
+
+    return { explicitlyMarked: false, isSafe: false };
+  }
+
+  private async getNonDefaultKibanaConfigs(
+    exposedConfigsToUsage: ExposedConfigsToUsage
+  ): Promise<ConfigUsageData> {
+    const config = await this.configService.getConfig$().pipe(first()).toPromise();
+    const nonDefaultConfigs = config.toRaw();
+    const usedPaths = await this.configService.getUsedPaths();
+    const exposedConfigsKeys = [...exposedConfigsToUsage.keys()];
+
+    return usedPaths.reduce((acc, usedPath) => {
+      const configValue = get(nonDefaultConfigs, usedPath);
+      const pluginId = exposedConfigsKeys.find((exposedConfigsKey) =>
+        usedPath.startsWith(exposedConfigsKey)
+      );
+      const { explicitlyMarked, isSafe } = this.getMarkedAsSafe(
+        exposedConfigsToUsage,
+        pluginId,
+        usedPath
+      );
+
+      // explicitly marked as safe
+      if (explicitlyMarked && isSafe) {
+        acc[usedPath] = configValue;
```

Comment:
should we stringify certain types like `ByteSizeValue` or `moment.Duration`?

Reply:
I'm getting raw config values in `configValue` which are exactly how the user have set in the configs. This means they can only be of primitive types or arrays of them.

I've added a check to see if the actual raw config is a moment duration to report its value instead of treating it as a string and sending `[redacted]` if not explicitly marked as safe.